src/hotspot/share/libadt/vectset.hpp

```diff
@@ -53,7 +53,9 @@ class VectorSet : public AnyObj {
 public:
   VectorSet();
   VectorSet(Arena* arena);
-  ~VectorSet() {}
+
+  // Allow move constructor for && (eg. capture return of function)
```

Comment:
It's not completely clear yet to me why this is required and how it correlates with `NONCOPYABLE` but I leave this to the experts :)

Reply:
I took this from @jcking . From what I understand:
`NONCOPYABLE` disables the copy constructor (`&`) and move operator. Somehow, this also disables the move constructor (`&&`). Re-enabling that one allows things like returning local containers, and capturing them via that move constructor.
```
Unique_Node_List some_function() {
  Unique_Node_List local_worklist;
  // do stuff
  return local_worklist;
}

void other_function() {
  Unique_Node_List capture_worklist = some_function();
  // capture_worklist has its scope widened to this function
}
```
But if someone has a more detailed explanation, I'm glad to hear it ;)

Reply:
pedantic: s/explicitly default it or define it/explicitly define it (possibly with a default definition)/.  But yes, that's
why the explicit definition is needed if you want to permit move  of noncopyable (e.g. move-only).